### PR TITLE
noise refactor part 2

### PR DESCRIPTION
**Pull request by efidoman**

### CODE REVIEW

File: bin/g5-calc-noise.sh [hunk 1]
```diff
@@ -28,73 +27,53 @@ function ReportNoiseAndExit()
...
 else
-  noise=0
+  noise=0.5
   ReportNoiseAndExit
 fi
 
 #    set initial x values based on date differences
 
 if [ $(bc <<< "$n < $MINRECORDS") -eq 1 ]; then
   # set noise = 0 - unknown
```
Comment (tynbendad) [inline bin/g5-calc-noise.sh]: update comment to "0.5 - light" ?

File: bin/g5-calc-noise.sh [hunk 2]
```diff
@@ -28,73 +27,53 @@ function ReportNoiseAndExit()
...
-  #echo "x delta=$x2x1Delta, y delta=$y2y1Delta" 
+  y2y1Delta=$(bc -l  <<< "${yarr[$i]} - ${yarr[$i-1]}")
+  if [ $(bc -l <<< "$y2y1Delta < 0") -eq 1 ]; then
+    y2y1Delta=$(bc -l <<< "0 - $y2y1Delta")
+  fi
+  remainder=$(bc -l <<< "$y2y1Delta - 10")
+  if [ $(bc -l <<< "$remainder > 0") -eq 1 ]; then
+    noise=$(bc -l <<< "$noise + $remainder/200") 
+  fi
+
```
Comment (tynbendad) [inline bin/g5-calc-noise.sh]: no more weighting for the most recent #'s?

File: bin/g5-calc-noise.sh (hunk 1 above)
Comment (efidoman) [reply bin/g5-calc-noise.sh]: Good point. I will do that and update the comments throughout to reflect the new calculation algorithm

File: bin/g5-calc-noise.sh (hunk 2 above)
Comment (efidoman) [reply bin/g5-calc-noise.sh]: Yes, I will add that back in to this algorithm.

**Comment by efidoman** [comment]
My latest noise unit test ....
```
./unit-test-actual-noise.sh
numRecords of 529 > 50 so limiting to 50
1562015328,144,149 [{"noise":0.04362, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562015628,142,146 [{"noise":0.03562, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562015928,140,143 [{"noise":0.02727, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562016228,137,140 [{"noise":0.02786, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562016528,132,138 [{"noise":0.05652, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562016828,126,134 [{"noise":0.07761, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562017128,118,129 [{"noise":0.11085, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562017429,111,123 [{"noise":0.12683, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562017728,106,115 [{"noise":0.10174, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562018028,103,109 [{"noise":0.07156, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562018328,100,104 [{"noise":0.05000, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562018628,99,101 [{"noise":0.02574, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562018928,97,99 [{"noise":0.02626, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562019229,94,98 [{"noise":0.05306, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562019529,90,96 [{"noise":0.08125, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562019830,83,92 [{"noise":0.12717, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562020129,73,87 [{"noise":0.20920, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562020428,64,79 [{"noise":0.24684, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562020729,58,70 [{"noise":0.22286, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562021030,54,62 [{"noise":0.16774, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562021331,51,55 [{"noise":0.09455, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562021930,53,51 [{"noise":0.08438, "noiseSend":1, "noiseString":"Clean","calculatedBy":"41minutes"}]
1562022229,55,52 [{"noise":0.09375, "noiseSend":1, "noiseString":"Clean","calculatedBy":"41minutes"}]
1562022529,59,54 [{"noise":0.12037, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562022830,62,57 [{"noise":0.11404, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562023128,65,60 [{"noise":0.12188, "noiseSend":1, "noiseString":"Clean","calculatedBy":"41minutes"}]
1562023428,68,63 [{"noise":0.13125, "noiseSend":1, "noiseString":"Clean","calculatedBy":"41minutes"}]
1562023728,70,66 [{"noise":0.07879, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562024028,76,69 [{"noise":0.13188, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562024328,82,73 [{"noise":0.16027, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562024628,86,78 [{"noise":0.13333, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562024929,93,84 [{"noise":0.13929, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562025228,101,90 [{"noise":0.15889, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562025528,110,97 [{"noise":0.17423, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562025832,116,104 [{"noise":0.15000, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
1562026130,121,112 [{"noise":0.10446, "noiseSend":1, "noiseString":"Clean","calculatedBy":"lastVariation"}]
root@eddie1:~/src/Logger/bin# 
```

**Pull request merged**